pkg/ccl/storageccl/export_storage_test.go

```diff
@@ -120,38 +127,83 @@ func TestPutHttp(t *testing.T) {
 	tmp, dirCleanup := testutils.TempDir(t)
 	defer dirCleanup()
 
-	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer r.Body.Close()
-		localfile := filepath.Join(tmp, filepath.Base(r.URL.Path))
-		switch r.Method {
-		case "PUT":
-			f, err := os.Create(localfile)
-			if err != nil {
-				http.Error(w, err.Error(), 500)
-				return
-			}
-			defer f.Close()
+	makeServer := func() (*url.URL, func() int, func()) {
+		var files int
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			defer r.Body.Close()
-			if _, err := io.Copy(f, r.Body); err != nil {
-				http.Error(w, err.Error(), 500)
-				return
+			localfile := filepath.Join(tmp, filepath.Base(r.URL.Path))
+			switch r.Method {
+			case "PUT":
+				f, err := os.Create(localfile)
+				if err != nil {
+					http.Error(w, err.Error(), 500)
+					return
+				}
+				defer f.Close()
+				defer r.Body.Close()
+				if _, err := io.Copy(f, r.Body); err != nil {
+					http.Error(w, err.Error(), 500)
+					return
+				}
+				files++
+			case "GET":
+				http.ServeFile(w, r, localfile)
+			case "DELETE":
+				if err := os.Remove(localfile); err != nil {
+					http.Error(w, err.Error(), 500)
+					return
+				}
+			default:
+				http.Error(w, "unsupported method "+r.Method, 400)
 			}
-		case "GET":
-			http.ServeFile(w, r, localfile)
-		case "DELETE":
-			if err := os.Remove(localfile); err != nil {
-				http.Error(w, err.Error(), 500)
-				return
-			}
-		default:
-			http.Error(w, "unsupported method "+r.Method, 400)
+		}))
+		t.Logf("Mock HTTP Storage %q", srv.URL)
+		uri, err := url.Parse(srv.URL)
+		if err != nil {
+			srv.Close()
+			t.Fatal(err)
+		}
+		return uri, func() int { return files }, srv.Close
+	}
+
+	t.Run("singleHost", func(t *testing.T) {
+		srv, files, cleanup := makeServer()
+		defer cleanup()
+		testExportToTarget(t, roachpb.ExportStorage{
+			Provider: roachpb.ExportStorageProvider_Http,
+			HttpPath: roachpb.ExportStorage_Http{BaseUri: srv.String() + "/"},
+		})
+		if wrote := files(); wrote != 11 {
+			t.Fatal("expected 11 files to be written to single http store")
+		}
+	})
+
+	t.Run("multiHost", func(t *testing.T) {
+		s1, f1, cleanup1 := makeServer()
+		defer cleanup1()
+		s2, f2, cleanup2 := makeServer()
+		defer cleanup2()
+		s3, f3, cleanup3 := makeServer()
```

Comment:
I feel like these would be clearer and only marginally more annoying to type if written as `srv3, files3, cleanup3`.